mrc/localization/locator.py

```diff
@@ -1,36 +1,51 @@
 import threading
-from dynaconf import settings
-import itertools
 import time
+from typing import Callable
 
 
 class Locator(threading.Thread):
-    def __init__(self):
+    def __init__(self, robot_list: list, read_data: Callable, process_data: Callable = None, interval: float = 0.0):
         threading.Thread.__init__(self)
-        self.locations = dict(itertools.product(settings.get('ROBOTS'), [None]))
-        self.data_reader = settings.get('DATA_READER')
-        self.data_processor = settings.get('DATA_PROCESSOR')
-        self.interval = settings.get('INTERVAL', 0)
+        self.locations = dict()
+        for robot_id in robot_list:
```

Comment:
Why not use previous code? Simply change settings.get('ROBOTS') to robot_list in dict(itertools.product(settings.get('ROBOTS'), [None])).

Reply:
It's a good solution, but Maybe monad (Optional) is begging "use me" 🙏 I can write such util if you like.